CDTDatastoreTests/CDTQIndexNameTests.m

```diff
@@ -0,0 +1,84 @@
+//
+//  CDTQIndexNameTests.m
+//  CDTDatastoreTests
+//
+//  Created by tomblench on 27/04/2018.
+//  Copyright © 2018 IBM Corporation. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import <XCTest/XCTest.h>
+#import "CloudantSyncTests.h"
+#import "CDTDatastore.h"
+#import "CDTDatastore+Query.h"
+#import "CDTDatastoreManager.h"
+#import "CDTDocumentRevision.h"
+#import "CDTQResultSet.h"
+
+// tests for "unusual" index names
+@interface CDTQIndexNameTests : CloudantSyncTests
+@end
+
+@implementation CDTQIndexNameTests
+
+// regression test for bug seen by customer when index name contains dashes and numbers
+- (void)testDashesAndNumbersInIndexName {
+    NSString *dbName = @"db";
+    NSString *indexName = @"flight-hv0999-20180426-ivl-ssa";
+    NSError *error;
+    CDTDatastore *ds = [self.factory datastoreNamed:dbName error:&error];
+    XCTAssertNil(error);
+    [ds ensureIndexed:@[@"type", @"number", @"name"] withName:indexName];
+    CDTDocumentRevision *rev = [CDTDocumentRevision revision];
+    rev.body = [NSMutableDictionary dictionaryWithDictionary:@{@"type":@"temp", @"number": @0, @"name": @"abc0"}];
+    [ds createDocumentFromRevision:rev error:&error];
+    XCTAssertNil(error);
+    CDTQResultSet *rs = [ds find:@{@"type":@"temp"} skip:0 limit:1000 fields:nil sort:@[@{@"number": @"desc"}]];
+    __block int resultCount = 0;
+    [rs enumerateObjectsUsingBlock:^(CDTDocumentRevision *rev, NSUInteger idx, BOOL *stop) {
+        resultCount++;
+    }];
+    XCTAssertTrue(resultCount == 1);
```

Comment:
Is it worth having any additional assertions here, or are we basically satisfied if there are no crashes/errors?